python/vineyard/deploy/local.py

```diff
@@ -148,4 +153,67 @@ def start_vineyardd(etcd_endpoints=None,
             etcd_ctx.__exit__(None, None, None)  # pylint: disable=no-member
 
 
+def init(socket=None, etcd_prefix=None, **kw):
+    '''
+    help new users to launch a local vineyardd and get a client as easy as possible
+
+    In a clean enviroment, simply use:
+    
+    .. code::
+    
+        vineyard.init()
+    
+    it will launch a local vineyardd and return a connected client to the vineyardd.
+
+    If the user has already lauched a vineyardd with a **SOCKET**. She can either set VINEYARD_IPC_SOCKET
+    as the SOCKET, and then run vineyard.init(), or simply use:
+    
+    .. code::
+    
+        vineyard.init(socket=SOCKET)
+
+    Sometimes we want to setup a local vineyard cluster consists of multiple vineyard daemons.
+    We can give the same **etcd_prefix** to each call of init:
+
+    .. code::
+    
+        client1 = vineyard.init(etcd_prefix='test')
+        client2 = vineyard.init(etcd_prefix='test')
+        client3 = vineyard.init(etcd_prefix='test')
+    
+    We will have three vineyard daemons running locally as a cluster to mimic the distributed scenario.      
+    '''
+    if socket is None:
+        socket = os.environ.get('VINEYARD_IPC_SOCKET', None)
+    try:
+        client = connect(socket)
+    except:
+        logger.info("Vineyard daemon on %s is not initialized", socket)
+        if etcd_prefix is not None:
+            kw['etcd_prefix'] = etcd_prefix
+            kw['etcd_endpoints'] = init.etcd_endpoints.get(etcd_prefix, None)
```

Comment:
It looks very strange to resolve `etcd_endpoints` by `etcd_prefix`... Why the later is the key to the former?